File: 1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n1 = word1.size();
        int n2 = word2.size();
        string ans = "";

        for (int i = 0; i < max(n1, n2); i++) {
            if (!word1.empty()) {
                ans += word1[0];
                word1.erase(0, 1);
            }
            if (!word2.empty()) {
                ans += word2[0];
                word2.erase(0, 1);
            }
        }

        return ans;
    }
};

File: 1894-merge-strings-alternately/merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function (word1, word2) {
    let ans = "";
    let n1 = word1.length, n2 = word2.length;

    for (let i = 0; i < Math.max(n1, n2); i++) {
        if (n1 > i) {
            ans += word1[i];
        }
        if (n2 > i) {
            ans += word2[i];
        }
    }

    return ans;
};